### PR TITLE
Extract logic for webroot into method and add test

### PR DESCRIPTION
**Pull request by kesselb**
Ref https://github.com/nextcloud/server/issues/10678

### CODE REVIEW

File: tests/lib/SetupTest.php [hunk 1]
```diff
@@ -130,4 +131,34 @@ public function testGetSupportedDatabaseException() {
...
+	 * @dataProvider findWebRootProvider
+	 * @param $url
+	 * @param $expected
+	 */
+	public function testFindWebRootCli($url, $expected) {
+		$this->config
+			->expects($this->once())
+			->method('getValue')
+			->will($this->returnValue($url));
+		\OC::$CLI = true;
```
Comment (blizzz) [inline tests/lib/SetupTest.php]: state is not restored after the test

File: lib/private/Setup.php [hunk 2]
```diff
@@ -431,27 +432,46 @@ private function pathToHtaccess() {
...
+	 * Find webroot from config
+	 *
+	 * @param SystemConfig $config
+	 * @return string
+	 * @throws InvalidArgumentException when invalid value for overwrite.cli.url
 	 */
-	public static function updateHtaccess() {
-		$config = \OC::$server->getSystemConfig();
-
+	public static function findWebRoot(SystemConfig $config): string {
```
Comment (nickvergessen) [inline lib/private/Setup.php]: should this even be public?

if you did this for better unit tests, you can use `$this->invokePrivate(\OC\Setup::class, 'findWebRoot', [$config])` in tests to run private/protected methods.

Comment (danielkesselberg) [reply lib/private/Setup.php]: I change the method to private. If the same logic is used elsewhere, we can make it public later.

**Review by rullzer — APPROVED** [review]
makes sense

**Review by blizzz — APPROVED** [review]

**Pull request merged**